### DISCUSSION TITLE
Auto update branch when no merge conflict

### DISCUSSION THREAD

**Discussion by makotoshimazu** [community/community#28874] [Pull Requests]
Hi,

We are using `Require branches to be up to date before merging` to prevent CI from failing after a merge.
Our regular workflow after getting merge approval is (1) hitting the `Update branch`, and (2) hitting `Enable auto merge`, and it works in most of the cases.

However, auto merge could be sometimes blocked repeatedly when merges happens frequently (like Renovate works hard and merges lots of PRs).
In this discussion, I'd like to propose `auto update branch` feature to mitigate this issue, which keeps the branch up-to-date to the tracking branch as much as possible.

Thanks!

**Comment by happy-barney** [comment]
maybe also option `Optimistic auto-merge` / `Serialize auto-merge` will be handy
- new PR CI jobs will start with changes "previous PR with auto-merge enabled" + "current PR", and so on
  - `auto-merge auto-update` should be able to configure to do `rebase` instead of `merge`
  - should start when all other checks are fulfilled - eg `require approval` 
- if one of previous PRs fails, whole PR queue will be restarted

**Comment by kevinh-csalabs** [comment]
Yes, and option to skip rerunning ci if there are no conflicts!
